src/api/app/models/project.rb

```diff
@@ -1586,7 +1586,7 @@ def self.remove_repositories(repositories, opts = {})
   end
 
   def has_remote_repositories?
-    repositories.any? { |r| r.download_repositories.any? }
+    DownloadRepository.where(repository_id: repositories.select(:id)).exists?
```

Comment:
You don't need the select here. Rails can make the connection repository id and create the same query like with the select.

Reply:
indeed